doc/make.jl

```diff
@@ -368,26 +368,60 @@ end
 
 const devurl = "v$(VERSION.major).$(VERSION.minor)-dev"
 
+function preRelease_versions(available_folders)
+    vnums = [VersionNumber(x) for x in available_folders]
+    master_version = maximum(vnums)
+    filter!(x -> x.major == 1 && x.minor == master_version.minor-1, vnums)
+    
+    pre_release_versions = []   # array to be returned
+
+    alpha_version = nothing
+    beta_version = nothing
+    rc_version = nothing
+
+    for version in vnums
+        prerelease = version.prerelease[1] 
+
+        if !isempty(prerelease)
+            if occursin(r"^rc", prerelease)
+                rc_version = version
+                beta_version = nothing  
+                alpha_version = nothing  
+            elseif occursin(r"^beta", prerelease)
+                if isnothing(rc_version)
+                    beta_version = version
+                end
+                alpha_version = nothing
+            elseif occursin(r"^alpha", prerelease)
+                if isnothing(rc_version) && isnothing(beta_version)
+                    alpha_version = version
+                end
+            end
+        end
+
+        if !isnothing(rc_version)
+            push!(pre_release_versions, "v$(rc_version)")
+        elseif !isnothing(beta_version)
+            push!(pre_release_versions, "v$(beta_version)")
+        elseif !isnothing(alpha_version)
+            push!(pre_release_versions, "v$(alpha_version)")
+        else
+            push!(pre_release_versions, "v$(master_version)")
+        end
+    end
+
+    return pre_release_versions
+end
+
```

Comment:
Now let's add a test set that checks something like:

```suggestion
@testset "preRelease_versions" begin
    @test preRelease_versions(["v1.0.0", "v1.1.0"]) == [...]
end
```

It should be a reasonably complete test set, testing the relevant edge cases, making sure that the function actually outputs what we expect. You probably need to also do `using Test` at the top of `make.jl`.

This testset will make `make.jl` exit with an error code if it fails, so we should be able to catch any regressions in the documentation builds.

Reply:
@mortenpi  sir just before I commit, the test case is 
`@testset "preRelease_versions" begin
    @test preRelease_versions(["v1.0.0", "v1.1.0"]) == [...]
end`
or
@testset "preRelease_versions" begin
    @test preRelease_versions(["v1.0.0", "v1.1.0"]) == ["v1.0.0", "v1.1.0"]
end

as it is pointing as  `invalid identifier` there at `[...]`

Reply:
The `[...]` was meant as a placeholder. The idea is to now add a bunch of test with expected outcome, trying to come up with edge cases, and then see if the function passes.

And I really think this is necessary here now, as I don't think the function is correct as it is... Here are some inputs to consider:
```
["v1.0.0", "v1.0.0-rc1", "v1.1.0"]
["v1.0.0", "v1.0.0-rc1", "v1.0.0-alpha1", "v1.1.0"]
["v1.0.0", "v1.0.0-alpha1", "v1.0.0-rc1", "v1.1.0"]
["v1.0.0-rc1", "v1.0.0-rc2", "v1.1.0"]
["v1.0.0-rc1", "v1.0.1-rc1", "v1.1.0"]
```